### PR TITLE
Added files for Minecraft v1.12

### PR DESCRIPTION
**Pull request by octan3**
Tested with current LiteLoader 1.12 and the final Minecraft Client v1.12

### CODE REVIEW

File: mcxray1.12/build.gradle [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            url = "http://files.minecraftforge.net/maven"
+        }
+        maven {
+            name = 'sponge'
+            url = 'http://repo.spongepowered.org/maven'
+        }
+        jcenter()
+    }
+    dependencies {
+        classpath 'net.minecraftforge.gradle:ForgeGradle:2.2-SNAPSHOT'
```
Comment (liach) [inline mcxray1.12/build.gradle]: Should be 2.3

**Comment by octan3** [comment]
Thanks @liach . I also added some missing fixes/files.

**Comment by uyjulian** [comment]
Thanks for the PR.
I'm in another country and only have access to my e-mail, so I'll
merge this whenever I can get back on my computer.

-- 
--
~Julian Uy

**Comment by Goobaroo** [comment]
@octan3, I think you're missing a commit in this pull.  I get a crash error after building this 1.12 version.

Error: java.lang.NoClassDefFoundError: bvm

**Comment by octan3** [comment]
@Goobaroo I downloaded my repo, edited the `settings.gradle` to:

```
include 'mcxray1.12', 'mcxrayshare'
```
and run `gradle build`. It created a working `.litemod` file in the folder `/mcxray1.12/build/libs/mod-mcxray1.12-1.0.0-mc1.12.litemod`.

**Comment by Goobaroo** [comment]
@octan3 Sorry, it wasn't on build, it was on run that there was an issue.

Tried grabbing the latest liteloader snapshot too from today.  But your build loaded fine for you?

[latest.txt](https://github.com/uyjulian/MinecraftX-RAY/files/1112082/latest.txt)

**Comment by uyjulian** [comment]
I'm back from abroad.
I will review the code tomorrow, since I'm jet-lagged.

**Comment by Goobaroo** [comment]
Welcome home.

**Comment by uyjulian** [comment]
Thank you @octan3

**Pull request merged**